connections/src/main/java/com/stripe/android/connections/model/serializer/PaymentAccountSerializer.kt

```diff
@@ -0,0 +1,27 @@
+package com.stripe.android.connections.model.serializer
+
+import com.stripe.android.connections.model.BankAccount
+import com.stripe.android.connections.model.LinkedAccount
+import com.stripe.android.connections.model.PaymentAccount
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.json.JsonContentPolymorphicSerializer
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
+
+internal object PaymentAccountSerializer :
+    JsonContentPolymorphicSerializer<PaymentAccount>(PaymentAccount::class) {
+
+    override fun selectDeserializer(element: JsonElement): KSerializer<out PaymentAccount> {
+        return when (element.objectValue) {
+            LinkedAccount.OBJECT -> LinkedAccount.serializer()
+            else -> BankAccount.serializer()
+        }
+    }
```

Comment:
@vardges-stripe @jameswoo-stripe - Update here - Instead of trying serializers, checking the `object` field on the json, that exists when it's a linked account. That way we avoid parsing twice and make this deterministic (if we know the object type and it's parser fails, there's something wrong in our model spec and we don't want to fallback to other deserializer)